providers/google/src/airflow/providers/google/cloud/operators/cloud_logging_sink.py

```diff
@@ -0,0 +1,490 @@
+            raise e
+
+
+class CloudLoggingUpdateSinkOperator(GoogleCloudBaseOperator):
+    """
+    Updates an existing Cloud Logging export sink.
+
+    :param sink_name: Required. Name of the sink to update.
+    :param project_id: Required. The ID of the Google Cloud project.
+    :param destination: New destination URI.
+    :param filter_: New filter expression for selecting log entries.
+    :param exclusion_filter: New exclusion filter. It will override old exclusion filter
+    :param description: New description for the sink.
+    :param disabled: Whether to disable/enable the sink.
+    :param bigquery_options: New bigquery related configuration.
+    :param include_children: Whether to export logs from child resources.
+    :param unique_writer_identity: If True, updates the writer identity.
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    """
+
+    template_fields: Sequence[str] = (
+        "sink_name",
+        "destination",
+        "filter_",
+        "exclusion_filter",
+        "project_id",
+        "description",
+        "gcp_conn_id",
+        "impersonation_chain",
+    )
+
+    def __init__(
+        self,
+        sink_name: str,
+        project_id: str,
+        destination: str | None = None,
+        filter_: str | None = None,
+        exclusion_filter: Sequence[dict] | dict | None = None,
+        description: str | None = None,
+        disabled: bool | None = None,
+        bigquery_options: dict | None = None,
+        include_children: bool | None = None,
+        unique_writer_identity: bool = False,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.sink_name = sink_name
+        self.project_id = project_id
+        self.destination = destination
+        self.filter_ = filter_
+        self.exclusion_filter = exclusion_filter
+        self.description = description
+        self.disabled = disabled
+        self.bigquery_options = bigquery_options
+        self.include_children = include_children
+        self.unique_writer_identity = unique_writer_identity
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+
+    def _validate_inputs(self):
```

Comment:
if we have some fields in common for verifying, I would suggest making one function and then override it with the parameters we need to check. But in general I see that mostly we are checking sink_name and project_id in the operators, so we can definitely move this outside the operator and just call it to not duplicate the code